lowRISC/opentitan
# Update lowrisc_ibex to lowRISC/ibex@b736680d

Update code from upstream repository
https://github.com/lowRISC/ibex.git to revision
b736680ddb73712c8e309b4d454d15e152ca3e33

* [dv] Add assertions checking double_fault_seen_o (Greg Chadwick)
* [dv] Add debug_mode to rvfi_ext (Greg Chadwick)
* [dv] Don't set rvfi_trap when executing ebreak into debug (Greg Chadwick)
* [dv] Fix iside error notification to cosim (Greg Chadwick)
* [fpv] Fix linting issues in oustanding access tracking logic (Greg Chadwick)
* [fpv] Add asserts to check crash dump connectivity (Greg Chadwick)
* Update google_riscv-dv to google/riscv-dv@d7c50c1 (Greg Chadwick)
* [rtl] Fix zbs bitmanip instruction tracer output (Greg Chadwick)
* [dv] fixed uvm check for mstatus.MPIE when interrupt is nmi (Saad Khalid)
* [dv,pmp] Put test done/fail sections before main for PMP tests (Marno van der Maas)
* [dv] added functional coverpoints (Saad Khalid)
* [dv] Generate random writes in custom CSRs (Canberk Topal)
* [dv] Add coverage for debug requests and interrupts while executing a dummy instruction (Andreas Kurth)
* [dv] Add coverpoints for dummy instructions in each stage (Andreas Kurth)
* [dv] V2S Coverage Implementation (Canberk Topal)
* [dv] Disable bad integrity on uninit accesses for mem error tests (Greg Chadwick)
* [cosim] Add write suppress support (Greg Chadwick)
* Fixup RVFI connection for pc_wdata (Canberk Topal)
* [cosim] Cosim integration of internal NMI (Canberk Topal)
* [ibex,dv] Add new test for memory integrity errors (Canberk Topal)
* [dv] Add probe and wait functionality for traps (Greg Chadwick)
* [dv] Add custom test done functionality (Greg Chadwick)
* [dv] Set rdata on write response in sequence not driver (Greg Chadwick)
* Fix-up new_seq_lib to start after stopping (Canberk Topal)
* [dv] Don't inject mem errors at test_control_addr (Canberk Topal)
* [ibex,dv] Switch to new sequence for memory errors (Canberk Topal)
* [dv] Don't set sync_exc_seen in Debug Mode (Canberk Topal)
* [dv] Add test glitching icache data or tag (Andreas Kurth)
* [dv] Create signal probe functions in `core_ibex_dut_probe_if` (Andreas Kurth)
* [dv] Add icache signals to `core_ibex_dut_probe_if` (Andreas Kurth)
* [dv] Add test glitching register file read data (Andreas Kurth)
* [dv] Create signal probe functions in `core_ibex_dut_probe_if` (Andreas Kurth)
* [dv] Add RF signals to `core_ibex_dut_probe_if` (Andreas Kurth)
* [dv] Compile `dv_assert.sv` before `prim_util_pkg.sv` (Andreas Kurth)
* [dv] Fix random data on uninit accesses (Greg Chadwick)
* [dv] Add missing isolation forks (Greg Chadwick)
* [xlm] Enable access to signals without dumping waves (Andreas Kurth)
* [doc] Add RF write enable glitch detection (Andreas Kurth)
* Increase iterations for PMP tests to improve coverage (Harry Callahan)
* Applied riscv-dv patch (Harry Callahan)
* Merge coverage using db's in a file (Harry Callahan)
* Move tests to their own directory, out/run/tests (Harry Callahan)
* Cleanup check_logs.py, remove redundant cosim_trace check (Harry Callahan)
* [dv] Add test glitching PC (Andreas Kurth)
* [dv] Add assert control for `NoAlertsTriggered` (Andreas Kurth)
* [dv] Add a matching NA4 in pmp_full_random_test (Canberk Topal)
* [dv] Randomize MPRV in pmp_full_random_test (Canberk Topal)
* [dv] Add directed instruction for random MSECCFG (Canberk Topal)
* [dv] Fix riscv_mem_error_test (Greg Chadwick)
* [dv] Access CPUCTRLSTS and SECURESEED in riscv_rand_instr_test (Greg Chadwick)
* [cov] Add illegal bin for misaligned data accesses (Greg Chadwick)
* [rtl] Fix dummy instructions (Greg Chadwick)
* Rework reset handling for UVM env (Harry Callahan)
* Break-out load_binary_to_mem for ISS and DUT seperately (Harry Callahan)
* [cosim] Fixup ebreak behaviour (Greg Chadwick)
* Increase delay between irq stimulus for nested_interrupt_test (Harry Callahan)
* Bump up timeout for irq_stimulus to accomodate fetch_enable changes (Harry Callahan)
* Change defaults for bad_intg on uninit accesses for Dmem/Imem (Harry Callahan)
* Add new uvm test to hit hardware breakpoints coverpoints (Harry Callahan)
* Redefine ECALL handler to no-longer jump to 'write_tohost:' (Harry Callahan)
* Generate test_done: and test_fail: sections using handshake mechanism (Harry Callahan)
* [rtl] Fix id_exception_o signal (Greg Chadwick)
* Update google_riscv-dv to google/riscv-dv@be9c75f (Harry Callahan)
* [dv,fcov] MPRV Effect Cross improvements (Canberk Topal)
* Extend illegal bin for None config in M-Mode (Canberk Topal)
* [cov] Remove ignored_csrs coverpoints (Greg Chadwick)
* [cov] Add waived CSRs IGNORED_CSRS (Greg Chadwick)
* [cov] Fix debug_wfi_cross (Greg Chadwick)
* [cov] Add some illegal bins related to instruction categories (Greg Chadwick)
* [cov] Remove pointless cross (Greg Chadwick)
* [rtl] Remove unused transition in ibex_controller FSM (Greg Chadwick)

Signed-off-by: Greg Chadwick <gac@lowrisc.org>
